server/services/team/service.ts

```diff
@@ -609,8 +608,9 @@ export class TeamService extends ServiceClass implements ITeamService {
 			}
 
 			this.TeamMembersModel.removeById(existingMember._id);
-
-			await this.unsubscribeFromMain(team.roomId, member.userId);
+			const removedUser = await this.Users.findOneById(member.userId, { projection: { _id: 1, username: 1 } });
+			const byUser = uid !== member.userId ? await this.Users.findOneById(uid, { projection: { _id: 1, username: 1 } }) : undefined;
```

Comment:
my only issue with this is that if a user is removing some other users, for every user being removed it will execute the same find for `uid`..

to avoid this I'd suggest to do a find for `uid` outside the loop and use it when needed.

Reply:
Following this line, it would be better to extract all the `userId`s before the loop, and then doing a find for all, instead of one by one 🤔 (then inside of this loop you can use `arr.find`)

Reply:
Great, I've just applied both changes (the find for `uid` is now performed only once, and I grouped the `findOneById` for each `userId` in a single `find`).